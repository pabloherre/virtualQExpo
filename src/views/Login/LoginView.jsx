import React, { Component } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import TextInput from '../../common/inputs/TextInput/TextInput';
import { Image, TouchableOpacity, View } from 'react-native';
import RoundedButton from '../../common/buttons/RoundedButton/RoundedButton';
import { setUser } from '../../modules/auth/Auth.actions';
import { safeArea } from '../../styles/common.styles';
import Typography from '../../common/typography/Typography';
import logo from '../../../assets/images/logo.png';

import AuthService from '../../modules/auth/Auth.service';
import UserService from '../../services/user/User.service';
import { showMessage } from 'react-native-flash-message';
import withBackground from '../../common/background/Background';

export class LoginView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    };
  }

  async componentDidMount() {
    const user = await AuthService.reAuthenticate();
    if (user) {
      UserService.setUser(user);
      this.props.navigation.navigate('Appointments');
    }
  }

  handleChange = (value, field) => {
    this.setState({ [field]: value });
  };

  handleLogin = async () => {
    try {
      const user = await AuthService.login(this.state);
      if (user) {
        await UserService.setUser(user);
        this.props.navigation.navigate('Appointments');
      }
    } catch (e) {
      showMessage({
        message: 'Login Failed',
        description: e.message,
        type: 'danger',
        icon: 'danger'
      });
    }
  };

  render() {
    const { email, password } = this.state;
    return (
      <View style={safeArea}>
        <View style={{ marginTop: 80 }}>
          <Image source={logo} style={{ alignSelf: 'center', marginTop: 10, marginBottom: 40, width: 350, resizeMode: 'contain' }} />
          <TextInput value={email} onChangeText={text => this.handleChange(text, 'email')} label="Email" />
          <TextInput value={password} secureTextEntry onChangeText={text => this.handleChange(text, 'password')} label="Password" />
          <View style={{ marginVertical: 20, alignItems: 'center' }}>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('Register')}>
              <Typography>
                Don't have an account?, register <Typography color="secondary">here!</Typography>
              </Typography>
            </TouchableOpacity>
          </View>
          <RoundedButton label="Login" onPress={this.handleLogin} isLoading={this.props.loading} />
        </View>
      </View>
    );
  }
}

function mapStateToProps(state) {
  return {
    loading: state.auth.loading
  };
}

const mapDispatchToProps = {
  setUser
};

export default compose(withBackground, connect(mapStateToProps, mapDispatchToProps))(LoginView);
