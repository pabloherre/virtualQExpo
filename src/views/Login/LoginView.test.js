import React from 'react';
import { LoginView } from './LoginView';
import { shallow, mount } from 'enzyme';
import AsyncStorage from '@react-native-community/async-storage';

const mockProps = {
  setUser: jest.fn(),
  navigation: {
    navigate: jest.fn()
  }
};

describe('<LoginView />', () => {
  it('should render correctly', async () => {
    const wrapper = mount(<LoginView />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should update email state when user types in', async () => {
    const wrapper = shallow(<LoginView />);
    wrapper.find('TextInput[label="Email"]').props().onChangeText('testEmail');
    expect(wrapper.state('email')).toBe('testEmail');
  });

  it('should update password state when user types in', async () => {
    const wrapper = shallow(<LoginView />);
    wrapper.find('TextInput[label="Password"]').props().onChangeText('testPassword');
    expect(wrapper.state('password')).toBe('testPassword');
  });

  it('should set user in async storage when log in', async () => {
    const wrapper = shallow(<LoginView {...mockProps} />);
    await wrapper.instance().handleLogin();

    expect(AsyncStorage.setItem).toHaveBeenCalled();
    expect(wrapper.instance().props.setUser).toHaveBeenCalled();
    expect(wrapper.instance().props.navigation.navigate).toHaveBeenCalled();
  });
});
