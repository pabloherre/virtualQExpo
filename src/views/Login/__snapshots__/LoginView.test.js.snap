// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginView /> should render correctly 1`] = `
<LoginView>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
        "paddingHorizontal": 40,
        "paddingVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "paddingHorizontal": 40,
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": 100,
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 100,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "alignSelf": "center",
                "marginVertical": 20,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                  "marginVertical": 20,
                }
              }
            />
          </Image>
          <TextInput
            label="Email"
            onChangeText={[Function]}
            value=""
          >
            <View
              style={
                Object {
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "paddingBottom": 10,
                    "paddingTop": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#0db4e4",
                      "fontSize": 20,
                      "fontWeight": "bold",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#0db4e4",
                        "fontSize": 20,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Email
                  </Text>
                </Text>
                <Component
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "borderBottomColor": "#0db4e4",
                      "borderBottomWidth": 1,
                      "fontSize": 20,
                      "height": 40,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                >
                  <TextInput
                    allowFontScaling={true}
                    onChangeText={[Function]}
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "borderBottomColor": "#0db4e4",
                        "borderBottomWidth": 1,
                        "fontSize": 20,
                        "height": 40,
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </Component>
              </View>
            </View>
          </TextInput>
          <TextInput
            label="Password"
            onChangeText={[Function]}
            secureTextEntry={true}
            value=""
          >
            <View
              style={
                Object {
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              <View
                style={
                  Object {
                    "paddingBottom": 10,
                    "paddingTop": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#0db4e4",
                      "fontSize": 20,
                      "fontWeight": "bold",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#0db4e4",
                        "fontSize": 20,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Password
                  </Text>
                </Text>
                <Component
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Object {
                      "borderBottomColor": "#0db4e4",
                      "borderBottomWidth": 1,
                      "fontSize": 20,
                      "height": 40,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                >
                  <TextInput
                    allowFontScaling={true}
                    onChangeText={[Function]}
                    rejectResponderTermination={true}
                    secureTextEntry={true}
                    style={
                      Object {
                        "borderBottomColor": "#0db4e4",
                        "borderBottomWidth": 1,
                        "fontSize": 20,
                        "height": 40,
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </Component>
              </View>
            </View>
          </TextInput>
          <View
            style={
              Object {
                "marginVertical": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "marginVertical": 20,
                }
              }
            >
              <ForwardRef
                onPress={[Function]}
              >
                <TouchableOpacity
                  hostRef={null}
                  onPress={[Function]}
                >
                  <ForwardRef(AnimatedComponentWrapper)
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        undefined,
                        Object {
                          "opacity": 1,
                        },
                      ]
                    }
                  >
                    <AnimatedComponent
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          undefined,
                          Object {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <Component>
                            <Typography
                              theme={
                                Object {
                                  "colors": Object {
                                    "background": "rgb(242, 242, 242)",
                                    "border": "rgb(216, 216, 216)",
                                    "card": "rgb(255, 255, 255)",
                                    "notification": "rgb(255, 59, 48)",
                                    "primary": "rgb(0, 122, 255)",
                                    "text": "rgb(28, 28, 30)",
                                  },
                                  "dark": false,
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "rgb(28, 28, 30)",
                                    "fontSize": 20,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "rgb(28, 28, 30)",
                                      "fontSize": 20,
                                    }
                                  }
                                >
                                  Dont have an account, register 
                                  <Component
                                    color="secondary"
                                  >
                                    <Typography
                                      color="secondary"
                                      theme={
                                        Object {
                                          "colors": Object {
                                            "background": "rgb(242, 242, 242)",
                                            "border": "rgb(216, 216, 216)",
                                            "card": "rgb(255, 255, 255)",
                                            "notification": "rgb(255, 59, 48)",
                                            "primary": "rgb(0, 122, 255)",
                                            "text": "rgb(28, 28, 30)",
                                          },
                                          "dark": false,
                                        }
                                      }
                                    >
                                      <Text
                                        style={
                                          Object {
                                            "color": undefined,
                                            "fontSize": 20,
                                          }
                                        }
                                      >
                                        <Text
                                          style={
                                            Object {
                                              "color": undefined,
                                              "fontSize": 20,
                                            }
                                          }
                                        >
                                          here
                                        </Text>
                                      </Text>
                                    </Typography>
                                  </Component>
                                </Text>
                              </Text>
                            </Typography>
                          </Component>
                          <PressabilityDebugView
                            color="cyan"
                          />
                        </View>
                      </View>
                    </AnimatedComponent>
                  </ForwardRef(AnimatedComponentWrapper)>
                </TouchableOpacity>
              </ForwardRef>
            </View>
          </View>
          <RoundedButton
            buttonStyle={Object {}}
            label="Login"
            onPress={[Function]}
            textStyle={Object {}}
          >
            <ForwardRef
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f77027",
                  "borderRadius": 30,
                  "flexDirection": "row",
                  "height": 60,
                  "justifyContent": "space-around",
                  "paddingHorizontal": 20,
                  "paddingVertical": 10,
                }
              }
            >
              <TouchableOpacity
                hostRef={null}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#f77027",
                    "borderRadius": 30,
                    "flexDirection": "row",
                    "height": 60,
                    "justifyContent": "space-around",
                    "paddingHorizontal": 20,
                    "paddingVertical": 10,
                  }
                }
              >
                <ForwardRef(AnimatedComponentWrapper)
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#f77027",
                        "borderRadius": 30,
                        "flexDirection": "row",
                        "height": 60,
                        "justifyContent": "space-around",
                        "paddingHorizontal": 20,
                        "paddingVertical": 10,
                      },
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                >
                  <AnimatedComponent
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#f77027",
                          "borderRadius": 30,
                          "flexDirection": "row",
                          "height": 60,
                          "justifyContent": "space-around",
                          "paddingHorizontal": 20,
                          "paddingVertical": 10,
                        },
                        Object {
                          "opacity": 1,
                        },
                      ]
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#f77027",
                          "borderRadius": 30,
                          "flexDirection": "row",
                          "height": 60,
                          "justifyContent": "space-around",
                          "opacity": 1,
                          "paddingHorizontal": 20,
                          "paddingVertical": 10,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#f77027",
                            "borderRadius": 30,
                            "flexDirection": "row",
                            "height": 60,
                            "justifyContent": "space-around",
                            "opacity": 1,
                            "paddingHorizontal": 20,
                            "paddingVertical": 10,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "white",
                              "fontSize": 20,
                              "paddingHorizontal": 10,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 20,
                                "paddingHorizontal": 10,
                              }
                            }
                          >
                            Login
                          </Text>
                        </Text>
                        <PressabilityDebugView
                          color="cyan"
                        />
                      </View>
                    </View>
                  </AnimatedComponent>
                </ForwardRef(AnimatedComponentWrapper)>
              </TouchableOpacity>
            </ForwardRef>
          </RoundedButton>
        </View>
      </View>
    </View>
  </View>
</LoginView>
`;
